portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,19 +27,44 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private List<String> names;
-
-  @Override
-  public void init() {
-      names = new ArrayList<>(Arrays.asList("Harry", "Ava", "Mia"));
-  }
-
+  private List<String> comments = new ArrayList<String>();
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Gson gson = new Gson();
-    String namesJson = gson.toJson(names);
+    String commentsJson = gson.toJson(comments);
 
     response.setContentType("application/json;");
-    response.getWriter().println(namesJson);
+    response.getWriter().println(commentsJson);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the comment from the form
+    String newComment = getNewComment(request);
+    if (newComment == "") {
```

Comment:
For comparing strings in Java, you'll ned to use `newComment.equals("")` - `==` does reference equality (i.e. same object) rather than value equality.